Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift

```diff
@@ -114,11 +116,24 @@ extension ARSwiftUIView: UIViewRepresentable {
 
 extension ARSwiftUIView {
     class Coordinator: NSObject, ARSCNViewDelegate, ARSessionDelegate {
+        private struct State: Sendable {
+            var onAddNodeAction: (@MainActor (SceneParameters) -> Void)?
+            var onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
+        }
+        
+        private let state = OSAllocatedUnfairLock(initialState: State())
```

Comment:
This is overkill. Instead, just refactor `Coordinator` to make all of its properties immutable. While you're at it, I suggest moving `Coordinator` into the previous extension (the one for `UIViewRepresentable`, since `Coordinator` satisfies a requirement of that protocol):

```swift
func updateUIView(_ uiView: ARViewType, context: Context) {}

func makeCoordinator() -> Coordinator {
    return .init(
        onDidChangeGeoTrackingStatusAction: onDidChangeGeoTrackingStatusAction,
        onCameraDidChangeTrackingStateAction: onCameraDidChangeTrackingStateAction,
        onDidUpdateFrameAction: onDidUpdateFrameAction,
        onAddNodeAction: onAddNodeAction,
        onUpdateNodeAction: onUpdateNodeAction
    )
}

class Coordinator: NSObject {
    let onDidChangeGeoTrackingStatusAction: ((ARSession, ARGeoTrackingStatus) -> Void)?
    let onCameraDidChangeTrackingStateAction: ((ARSession, ARCamera.TrackingState) -> Void)?
    let onDidUpdateFrameAction: ((ARSession, ARFrame) -> Void)?
    let onAddNodeAction: (@MainActor (SceneParameters) -> Void)?
    let onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
    
    init(
        onDidChangeGeoTrackingStatusAction: ((ARSession, ARGeoTrackingStatus) -> Void)?,
        onCameraDidChangeTrackingStateAction: ((ARSession, ARCamera.TrackingState) -> Void)?,
        onDidUpdateFrameAction: ((ARSession, ARFrame) -> Void)?,
        onAddNodeAction: (@MainActor (SceneParameters) -> Void)?,
        onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
    ) {
        self.onDidChangeGeoTrackingStatusAction = onDidChangeGeoTrackingStatusAction
        self.onCameraDidChangeTrackingStateAction = onCameraDidChangeTrackingStateAction
        self.onDidUpdateFrameAction = onDidUpdateFrameAction
        self.onAddNodeAction = onAddNodeAction
        self.onUpdateNodeAction = onUpdateNodeAction
    }
}
```

and then conforming it to `ARSCNViewDelegate` and `ARSessionDelegate` in extensions:

```swift
extension ARSwiftUIView.Coordinator: ARSCNViewDelegate {
    func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
        let sceneParameters = SceneParameters(renderer: renderer, node: node, anchor: anchor)
        Task { @MainActor [onAddNodeAction] in
            onAddNodeAction?(sceneParameters)
        }
    }
    
    func renderer(_ renderer: SCNSceneRenderer, didUpdate node: SCNNode, for anchor: ARAnchor) {
        let sceneParameters = SceneParameters(renderer: renderer, node: node, anchor: anchor)
        Task { @MainActor [onUpdateNodeAction] in
            onUpdateNodeAction?(sceneParameters)
        }
    }
}

extension ARSwiftUIView.Coordinator: ARSessionDelegate {
    func session(_ session: ARSession, didChange geoTrackingStatus: ARGeoTrackingStatus) {
        onDidChangeGeoTrackingStatusAction?(session, geoTrackingStatus)
    }
    
    func session(_ session: ARSession, cameraDidChangeTrackingState camera: ARCamera) {
        onCameraDidChangeTrackingStateAction?(session, camera.trackingState)
    }
    
    func session(_ session: ARSession, didUpdate frame: ARFrame) {
        onDidUpdateFrameAction?(session, frame)
    }
}
```

Note that `MainActor.run` isn't necessary either.

Reply:
Applied, thank you!